### PR TITLE
Ics20 withdraw action

### LINKED ISSUE #982: Define a `Withdraw` transaction action
Inbound ICS20 transfers happen when relayers post IBC packets to the chain. Outbound transfers are initiated on Penumbra, and need to be integrated with the Penumbra transaction model.

To do this, we should add `Withdraw` transaction action that specifies:

- a destination chain;
- a transparent value, consisting of an amount + asset id (this implies the denom, which implies the channel/port/etc);
- a destination chain address;
- a "sender" penumbra address to use to return funds;
- a timeout height (on the source [Penumbra] chain ?);
- a timeout time.

This action should consume the value from the transaction's value balance and burn it.

We need to include the sender address to be able to return funds in the event of a transfer failure, but ideally this should be an uncommon case, and ideally outbound transfers should not disclose the user's Penumbra address.  To handle this, we should also

- [x] #983
- [x] #984

and then generate an ephemeral address to put in the `Withdraw` action. We have 11-byte address indices, so we have 2^88 - 2^64 ~ 2^88 possible ephemeral addresses; the birthday bound suggests we're likely to get a collision only after 2^44 IBC withdrawals, which seems fine.  However, from the point of view of the `Withdraw` action, the fact that the address _was_ actually generated this way is irrelevant (since the ephemeral addresses are publicly indistinguishable from normal addresses), so it doesn't affect the transaction action at all.

### PR DESCRIPTION
**Pull request by aubrika**
Closes #982

### CODE REVIEW

**Comment by avahowell** [comment]
LGTM pending those two comments

File: proto/proto/ibc.proto
Comment (hdevalence) [inline proto/proto/ibc.proto:61]: Elsewhere we use `uint64` heights, is there a reason to use the IBC proto type in our transaction action?

File: proto/proto/ibc.proto
Comment (hdevalence) [inline proto/proto/ibc.proto:54]: Since this has a different meaning than normal, we should rename it, perhaps to `return_address`, since that's what it's used for?

File: proto/proto/transaction.proto
Comment (hdevalence) [inline proto/proto/transaction.proto:57]: We should add in the action now, and reject it until the implementation is ready.  Otherwise we don't have a good way to incrementally implement more and more functionality, since the datatypes aren't there yet.

File: transaction/src/action.rs
Comment (hdevalence) [inline transaction/src/action.rs:51]: ditto https://github.com/penumbra-zone/penumbra/pull/1261/files#r946167248

File: transaction/src/action.rs
Comment (hdevalence) [inline transaction/src/action.rs:176]: probably no particular reason to move this out of the usual place?

File: transaction/src/action/ibc.rs
Comment (hdevalence) [reply transaction/src/action/ibc.rs:37]: Yes, this is correct -- I'd tweak some of the wording in the comment though: it consumes *from* the transaction's value balance, rather than consuming the value balance itself.

**Review by hdevalence — CHANGES_REQUESTED** [review]
We should add the new withdrawal action into the `Action` enum now, so that it becomes part of our data model.

Because transfers aren't implemented yet, we can reject them in the IBC component, using a similar method as we use to reject LP actions: https://github.com/penumbra-zone/penumbra/blob/main/component/src/dex/component.rs#L31-L44

File: transaction/src/action.rs
Comment (aubrika) [reply transaction/src/action.rs:176]: I thought so too, however:

```error[E0046]: not all trait items implemented, missing: `Error`
   --> transaction/src/action.rs:136:1
    |
136 | impl TryFrom<pb::Action> for Action {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `Error` in implementation
    |
    = help: implement the missing item: `type Error = Type;`

For more information about this error, try `rustc --explain E0046`.
error: could not compile `penumbra-transaction` due to previous error```

File: proto/proto/ibc.proto
Comment (avahowell) [reply proto/proto/ibc.proto:61]: IBC heights are slightly different- they contain both an epoch and a height. We don't have epochs so the epoch is always 0. Using an IBC height makes it easier to interoperate with other chains, but if this is only going to be used for penumbra heights then we can just use a uint64

File: transaction/Cargo.toml
Comment (hdevalence) [inline transaction/Cargo.toml:24]: This will need to be rebased on `main` before merging, so it doesn't touch the Poseidon parameters.

**Review by hdevalence — APPROVED** [review]
LGTM but these changes need to be rebased on `main` to deconflict the edits to the Poseidon crate.

**Pull request merged**
